docs/specs/governance_v2_flows.puml

```diff
@@ -0,0 +1,289 @@
+endbox
+
+box "(offchain)" #Coral
+database bitdaogovernance as bitdaogovernance
+end box
+
+box "(onchain)" #Lavender
+collections Executor as executor
+collections Token as token
+collections BondContract as bondcontract
+collections RewardsContract as rewardscontract
+collections InsuranceContract as insurancecontract
+collections GrantsTreasury as grantstreasury
+collections BitDAOTreasury as bitdaotreasury
+collections BitDAOGovernanceV2 as bitdaogovernance
+end box
+
+autonumber "Current:[00]"
+
+== Current ==
+
+alice -> grantstreasury: create grants treasury
+alice <-- grantstreasury: treasury address
+
+alice -> forum: create forum proposal
+activate forum
+community -> forum: get proposals
+community <-- forum: list of proposals
+community -> forum: put positive/negative feedback
+alice <-- forum: get positive/negative feedback
+deactivate forum
+
+community -> community: delegate votes
+
+alice -> bitdaogovernance: submit treasury role proposal
+alice <-- bitdaogovernance: success/fail if alice has treasury role
+tokenholders -> bitdaogovernance: vote on proposal
+tokenholders <-- bitdaogovernance: treasury role vote confirmation
+
+bitdaoadmin -> bitdaogovernance: get Alice's proposal result
+bitdaoadmin <-- bitdaogovernance: Alice's proposal pass/fail result
+
+bitdaoadmin -> bitdaogovernance: execute proposal
+bitdaogovernance -> bitdaotreasury: transfer funds to grants treasury
+bitdaotreasury -> grantstreasury: funds transfer
+
+alt#Gold #DarkSeaGreen trace
+  alice <-- grantstreasury: transfer confirmation
+end
+
+alice --> alice: execute proposal
+
+newpage
+
+autonumber "Bond:[00]"
+
+== Bond ==
+
+
+alice -> grantstreasury: create grants treasury
+alice <-- grantstreasury: treasury address
+
+' bond
+
+alice -> forum: create forum proposal
+activate forum
+guarantor -> forum: get proposals
+guarantor <-- forum: list of proposals
+guarantor -> forum: put positive/negative feedback
+
+alice -> bondcontract: create bond
+guarantor -> bondcontract: contribute BIT tokens
+guarantor <-- bondcontract: ERC20 Fungible bond tokens (1:1 swap)
+
+' treasury
+
+community -> forum: get proposals
+community <-- forum: list of proposals
+community -> forum: put positive/negative feedback
+alice <-- forum: get positive/negative feedback
+deactivate forum
+
+community --> community: delegate votes
+
+alice -> bitdaogovernance: submit treasury proposal
+alice <-- bitdaogovernance: success/fail if alice has treasury role
+tokenholders -> bitdaogovernance: vote on proposal
+tokenholders <-- bitdaogovernance: treasury role vote confirmation
+
+bitdaoadmin -> bitdaogovernance: get vote result
+bitdaoadmin <-- bitdaogovernance: pass/fail proposal result
+bitdaoadmin -> bitdaogovernance: execute proposal
+bitdaogovernance -> bitdaotreasury: allow funds transfer
+bitdaotreasury -> grantstreasury: transfer funds
+
+' execution
+
+alice --> alice: execute proposal
+
+
+' tranche request
+
+'alice -> forum: create tranche release request forum proposal
```

Comment:
suggestion (blocking): no checking in of commented out code.

Generally when code / documentation is commented out, then it does not belong in a commit. It belong either in a stash, or live guard by a feature flag.
In this case deletion is the way forward.

Reply:
It updated flows with on-chain governance instead of snapshot, but I decided to have them in separate file. Maybe we should rename flows.puml to snapshot_flows.puml and remove any on-chain governance flows from it or switch them to snapshot ones (like insurance and rewards there). @shahankhatch should also comment on it.